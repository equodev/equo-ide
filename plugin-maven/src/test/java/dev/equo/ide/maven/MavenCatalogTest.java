/*******************************************************************************
 * Copyright (c) 2022-2025 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.ide.maven;

import com.diffplug.selfie.StringSelfie;
import java.io.IOException;
import org.junit.jupiter.api.Test;

public class MavenCatalogTest extends MavenHarness {
	private StringSelfie test(String content) throws IOException, InterruptedException {
		setPom(
				content
						+ "\n"
						+ "<filters>\n"
						+ "  <filter><platformNone>true</platformNone></filter>\n"
						+ "</filters>\n");
		return mvnw("equo-ide:list -Drequest")
				.snapshotBetween("(default-cli) @ equo-maven-test-harness ---", "[INFO] BUILD SUCCESS");
	}

	@Test
	public void simple() throws Exception {
		test("<platform><version>4.27</version></platform><jdt/>").toMatchDisk("jdt");
		test("<platform><version>4.27</version></platform><gradleBuildship/>")
				.toMatchDisk("gradleBuildship");
	}

	@Test
	public void versionOverride() throws IOException, InterruptedException {
		test("<jdt><version>4.25</version></jdt>").toMatchDisk("jdt-spec");
		test("<platform/><jdt><version>4.25</version></jdt>").toMatchDisk("platform-neutral-jdt-spec");
		test("<platform><version>4.25</version></platform><jdt/>").toMatchDisk("platform-spec");
		test("<platform><version>4.25</version></platform><jdt><version>4.25</version></jdt>")
				.toMatchDisk("both-spec");
	}

	@Test
	public void urlOverride() throws IOException, InterruptedException {
		test("<jdt><url>http://test.url/</url></jdt>").toMatchDisk();
	}
}
