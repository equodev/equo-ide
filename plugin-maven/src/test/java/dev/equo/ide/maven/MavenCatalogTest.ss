╔═ simple/gradleBuildship ═╗
+-------------+-----------------------------------------------------------------------------------------+
| kind        | value                                                                                   |
+-------------+-----------------------------------------------------------------------------------------+
| p2repo      | https://download.eclipse.org/buildship/updates/e431/releases/3.x/3.1.10.v20240802-1211/ |
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/                                      |
| install     | org.eclipse.buildship.feature.group                                                     |
| install     | org.eclipse.platform.ide.categoryIU                                                     |
| install     | org.eclipse.releng.java.languages.categoryIU                                            |
| filter      | maven-filter-unnamed--1003173200                                                        |
|   osgi.arch | dont-include-platform-specific-artifacts                                                |
|   osgi.os   | dont-include-platform-specific-artifacts                                                |
|   osgi.ws   | dont-include-platform-specific-artifacts                                                |
+-------------+-----------------------------------------------------------------------------------------+

[INFO] ------------------------------------------------------------------------
╔═ simple/jdt ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
╔═ urlOverride ═╗
+-------------+----------------------------------------------+
| kind        | value                                        |
+-------------+----------------------------------------------+
| p2repo      | http://test.url/                             |
| install     | org.eclipse.platform.ide.categoryIU          |
| install     | org.eclipse.releng.java.languages.categoryIU |
| filter      | maven-filter-unnamed--1003173200             |
|   osgi.arch | dont-include-platform-specific-artifacts     |
|   osgi.os   | dont-include-platform-specific-artifacts     |
|   osgi.ws   | dont-include-platform-specific-artifacts     |
+-------------+----------------------------------------------+

[INFO] ------------------------------------------------------------------------
╔═ versionOverride/both-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
╔═ versionOverride/jdt-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
╔═ versionOverride/platform-neutral-jdt-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
╔═ versionOverride/platform-spec ═╗
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
╔═ [end of file] ═╗
