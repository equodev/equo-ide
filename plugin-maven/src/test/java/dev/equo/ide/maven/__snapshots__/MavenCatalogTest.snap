dev.equo.ide.maven.MavenCatalogTest.simple[gradleBuildship]=[
+-------------+-----------------------------------------------------------------------------------------+
| kind        | value                                                                                   |
+-------------+-----------------------------------------------------------------------------------------+
| p2repo      | https://download.eclipse.org/buildship/updates/e431/releases/3.x/3.1.10.v20240802-1211/ |
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/                                      |
| install     | org.eclipse.buildship.feature.group                                                     |
| install     | org.eclipse.platform.ide.categoryIU                                                     |
| install     | org.eclipse.releng.java.languages.categoryIU                                            |
| filter      | maven-filter-unnamed--1003173200                                                        |
|   osgi.arch | dont-include-platform-specific-artifacts                                                |
|   osgi.os   | dont-include-platform-specific-artifacts                                                |
|   osgi.ws   | dont-include-platform-specific-artifacts                                                |
+-------------+-----------------------------------------------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]


dev.equo.ide.maven.MavenCatalogTest.simple[jdt]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.27/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]


dev.equo.ide.maven.MavenCatalogTest.urlOverride=[
+-------------+----------------------------------------------+
| kind        | value                                        |
+-------------+----------------------------------------------+
| p2repo      | http://test.url/                             |
| install     | org.eclipse.platform.ide.categoryIU          |
| install     | org.eclipse.releng.java.languages.categoryIU |
| filter      | maven-filter-unnamed--1003173200             |
|   osgi.arch | dont-include-platform-specific-artifacts     |
|   osgi.os   | dont-include-platform-specific-artifacts     |
|   osgi.ws   | dont-include-platform-specific-artifacts     |
+-------------+----------------------------------------------+

[INFO] ------------------------------------------------------------------------
]


dev.equo.ide.maven.MavenCatalogTest.versionOverride[both-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]


dev.equo.ide.maven.MavenCatalogTest.versionOverride[jdt-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]


dev.equo.ide.maven.MavenCatalogTest.versionOverride[platform-neutral-jdt-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]


dev.equo.ide.maven.MavenCatalogTest.versionOverride[platform-spec]=[
+-------------+----------------------------------------------------+
| kind        | value                                              |
+-------------+----------------------------------------------------+
| p2repo      | https://download.eclipse.org/eclipse/updates/4.25/ |
| install     | org.eclipse.platform.ide.categoryIU                |
| install     | org.eclipse.releng.java.languages.categoryIU       |
| filter      | maven-filter-unnamed--1003173200                   |
|   osgi.arch | dont-include-platform-specific-artifacts           |
|   osgi.os   | dont-include-platform-specific-artifacts           |
|   osgi.ws   | dont-include-platform-specific-artifacts           |
+-------------+----------------------------------------------------+

[INFO] ------------------------------------------------------------------------
]